pkg/controller/worker/machine_dependencies.go

```diff
@@ -14,12 +14,281 @@
+		return fmt.Errorf("failed to get shoot nodes: %v", err)
+	}
+
+	// go through each node, for each one without the right annotation, get the private network
+	var targetCidrs []string
+	for _, n := range shootNodes.Items {
+		if n.Annotations == nil || n.Annotations[equinixMetalPrivateNetworkAnnotations] == "" {
+			// we didn't have it, so get it from the Equinix Metal API, and save it
+			deviceID, err := deviceIDFromProviderID(n.Spec.ProviderID)
+			if deviceID == "" || err != nil {
+				continue
+			}
+			nodePrivateNetwork, err := getNodePrivateNetwork(deviceID, c, ns)
+			if err != nil {
+				return fmt.Errorf("error getting private network from Equinix Metal API for %s: %v", n.Spec.ProviderID, err)
+			}
+			if nodePrivateNetwork == "" {
+				continue
+			}
+			if n.Annotations == nil {
+				n.Annotations = map[string]string{}
+			}
+			// if it was not set already, set it and save it
+			if n.Annotations[equinixMetalPrivateNetworkAnnotations] != nodePrivateNetwork {
+				n.Annotations[equinixMetalPrivateNetworkAnnotations] = nodePrivateNetwork
+				// update the node in the k8s cluster ***
+				patch, _ := json.Marshal(map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"annotations": n.Annotations,
+					},
+				})
+				if err := shoot.Patch(ctx, &n, client.RawPatch(types.StrategicMergePatchType, patch)); err != nil {
+					return fmt.Errorf("unable to patch node %s with private network cidr: %v", n.Name, err)
+				}
+			}
+		}
+		targetCidrs = append(targetCidrs, n.Annotations[equinixMetalPrivateNetworkAnnotations])
+	}
+
+	// sort them for consistency; the order really doesn't matter, as long as it is consistent
+	targetCidrs = unique(targetCidrs)
+	sort.Strings(targetCidrs)
+	cidrs := strings.Join(targetCidrs, ",")
+
+	// get the configmap
+	cm := &corev1.ConfigMap{}
+	if err := c.Get(ctx, client.ObjectKey{
+		Namespace: w.cluster.ObjectMeta.GetNamespace(),
+		Name:      vpnSeedConfigMap,
+	}, cm); err != nil {
+		return fmt.Errorf("failed to get configmap: %v", err)
+	}
+
+	// we have the configmap, update it with the right CIDR for nodes
+	existing, ok := cm.Data[nodeNetwork]
+	// if it already exists, check if we are updating it
+	if ok {
+		// check for a conflict
+		if existing != cidrs {
+			return fmt.Errorf("conflicting cidr, existing %s replacement %s", existing, cidrs)
+		}
+		// matched, so nothing left to do
+		return nil
+	}
+	cm.Data[nodeNetwork] = cidrs
+	return c.Update(ctx, cm)
+}
+
+func unique(stringSlice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+	for _, entry := range stringSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
+
+// getNodePrivateNetwork use the Equinix Metal API to get the CIDR of the private network given a providerID.
+func getNodePrivateNetwork(deviceID string, kClient client.Client, namespace string) (string, error) {
+	ctx := context.Background()
```

Comment:
Don't use `context.Background`, rather pass the `ctx` from `CleanupMachineDependencies`

Reply:
Done